app/controllers/api/v1/posts_controller.rb

```diff
@@ -20,12 +20,28 @@ def create
         @post.save if validate_post(@user, @post)
       end
 
+      def update
+        @post = Post.find(params[:id])
+        @post.update(params.require(:post).permit(:title, :body))
+        if validate_not_empty_attriutes(@post)
+          @post.save
+          render 'api/v1/posts/create', formats: [:json], locals: { post: @post }
+        end
```

Comment:
The save here is not doing anything, as changes would have been saved in the update. The validations should be run before the update.

Also, the `params.require(:post).permit(:title, :body)` should be moved to a function as you are also using it in the create